### ISSUE TITLE
Add complete SMuFL brass techniques range

### ISSUE THREAD

**Issue by mdgood** [w3c/musicxml#44]
MusicXML could add support for the complete SMuFL brass techniques range U+E5D0–U+E5EF for better interoperability between applications. The range is documented at http://w3c.github.io/smufl/gitbook/tables/brass-techniques.html.

This would involve:
- Adding different lengths of the scoop, plop, doit, and falloff elements to cover the three different lengths provided within SMuFL 1.19. MusicXML distinguishes some other SMuFL symbols by the line-type and line-shape attributes rather than by element name.
- Defining the MusicXML shake element to refer to the jazz turn symbol U+E5E4.
- Defining the MusicXML stopped element to refer to the muted (closed) symbol U+E5E5.

It would also include adding the following SMuFL symbols in this range that are not directly represented in MusicXML 3.0 as of SMuFL 1.19:
- Flip (U+E5E1)
- Smear (U+E5E2)
- Bend (U+E5E3)
- Half muted (U+E5E6)
- Open (U+E5E7)
- Harmon mute symbols (U+E5E8, U+E5E9, U+E5EA, and U+E5EB)
- Valve trill (U+E5EF)

David Webber requested coverage for the symbols listed in Example 24-11 of Gardner Read's 1979 book _Music Notation_ on the MusicXML forum: http://forums.makemusic.com/viewtopic.php?f=12&t=2179#p5551. This includes the flip and bend symbols listed above.

See the MusicXML forum discussion started by George Litterst about the definition of the MusicXML 3.0 shake element at http://forums.makemusic.com/viewtopic.php?f=12&t=2431.

**Comment by mdgood** [comment]
For the scoop, plop, doit, and falloff elements, we can add a new "line-length" attribute group to the empty-line complex type:
```
	<xs:simpleType name="line-length">
		<xs:restriction base="xs:token">
			<xs:enumeration value="short"/>
			<xs:enumeration value="medium"/>
			<xs:enumeration value="long"/>
		</xs:restriction>
	</xs:simpleType>

	<xs:attributeGroup name="line-length">
		<xs:attribute name="line-length" type="line-length"/>
	</xs:attributeGroup>

	<xs:complexType name="empty-line">
		<xs:attributeGroup ref="line-shape"/>
		<xs:attributeGroup ref="line-type"/>
		<xs:attributeGroup ref="line-length"/>
		<xs:attributeGroup ref="dashed-formatting"/>
		<xs:attributeGroup ref="print-style"/>
		<xs:attributeGroup ref="placement"/>
	</xs:complexType>
```

The harmon mute can be represented similarly to the hole element in MusicXML 3.0. The harmon-closed type is defined like the current MusicXML 3.0 hole-closed type, except with "harmon" instead of "hole" in all the relevant names:
```
	<xs:simpleType name="harmon-closed-location">
		<xs:restriction base="xs:string">
			<xs:enumeration value="right"/>
			<xs:enumeration value="bottom"/>
			<xs:enumeration value="left"/>
			<xs:enumeration value="top"/>
		</xs:restriction>
	</xs:simpleType>

	<xs:simpleType name="harmon-closed-value">
		<xs:restriction base="xs:string">
			<xs:enumeration value="yes"/>
			<xs:enumeration value="no"/>
			<xs:enumeration value="half"/>
		</xs:restriction>
	</xs:simpleType>

	<xs:complexType name="harmon-closed">
		<xs:simpleContent>
			<xs:extension base="harmon-closed-value">
				<xs:attribute name="location" type="harmon-closed-location"/>
			</xs:extension>
		</xs:simpleContent>
	</xs:complexType>

	<xs:complexType name="harmon-mute">
		<xs:sequence>
			<xs:element name="harmon-closed" type="harmon-closed"/>
		</xs:sequence>
		<xs:attributeGroup ref="print-style"/>
		<xs:attributeGroup ref="placement"/>
	</xs:complexType>
```

The flip, smear, and bend elements can be represented by new `<flip>`, `<smear>`, and `<brass-bend>` elements that are new children of the `<technical>` element. The brass-bend name avoids conflict with the existing `<bend>` element for guitar bends.

It is unclear how to handle the remaining open, half-muted, and valve trill symbols. These could be left to the other-technical element. Since the open element looks like an open hole symbol in wind techniques, perhaps we could add a smufl attribute to the hole element.

Comments and suggestions are most welcome!

**Comment by mdgood** [comment]
My preference at this point is to add new `<open>` and `<half-muted>` elements while leaving the valve trill symbol to the other-technical element. The first two symbols seem more commonly used, including being referenced in _Behind Bars_. Although sharing some appearance with the hole or harmonic elements, those have many extra semantic elements that do not seem appropriate for sharing, in contrast with the `<stopped>` element. Both of these new elements would have an empty-placement-smufl type, similar to the `<stopped>` element.

**Comment by mdgood** [comment]
Pull request #196 addresses this issue.

**Comment by mdgood** [comment]
Pull request #207 fixes a bug from work on issue #44 that affects this issue too.
